provisioning/service/src/Contract/ContractApiHttp.cs

```diff
@@ -149,22 +149,22 @@ internal class ContractApiHttp : IContractApiHttp
                     ReadOnlyCollection<Exception> innerExceptions = ex.Flatten().InnerExceptions;
                     if (innerExceptions.Any(e => e is TimeoutException))
                     {
-                        throw new ProvisioningServiceClientTransportException(ex.Message, ex);
+                        throw new DeviceProvisioningServiceException(ex.Message, ex, true);
                     }
 
                     throw;
                 }
                 catch (TimeoutException ex)
                 {
-                    throw new ProvisioningServiceClientTransportException(ex.Message, ex);
+                    throw new DeviceProvisioningServiceException(ex.Message, ex, true);
                 }
                 catch (IOException ex)
                 {
-                    throw new ProvisioningServiceClientTransportException(ex.Message, ex);
+                    throw new DeviceProvisioningServiceException(ex.Message, ex, true);
                 }
                 catch (HttpRequestException ex)
                 {
-                    throw new ProvisioningServiceClientTransportException(ex.Message, ex);
+                    throw new DeviceProvisioningServiceException(ex.Message, ex, true);
                 }
                 catch (TaskCanceledException ex)
```

Comment:
We might not need this catch. TaskCanceledException inherits from OperationCanceledException.

Reply:
According to the comment on L171, TaskCanceledException is thrown when HttpClient times out, I believe that's why we catch `TaskCanceledException` here and check if `cancellationToken.IsCancellationRequested` inside.

Reply:
I see. Makes sense.